cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/config/ETLBatchConfig.java

```diff
@@ -29,34 +31,49 @@
 public final class ETLBatchConfig extends ETLConfig {
   private final String schedule;
   private final List<ETLStage> actions;
+  private final List<LocalizeResourceInfo> resourcesToLocalize;
```

Comment:
I just remembered another reason for not doing this at an individual plugin level. Ideally, an operation like localizing a resource would be done in the client side (beforeSubmit). However, today, transforms do not have a ``prepareRun`` stage, which is the code that is executed in ``beforeSubmit`` for sources and sinks. Unsure if this is a shortcoming of the current API, or if it is by design.

@anew: Your point is also valid. If we did it the way it is done in this PR currently (pipeline config), then that would mean the different plugins somehow need to access the pipeline config (or hack around by expecting it as a field in the plugin's config).

@chtyim: I will think about the approach you suggested. However, before that is implemented, we would need a ``prepareRun`` method in Transforms too? Doesn't make sense to localize resources in every task (in the initialize method)?